File: 104-maximum-depth-of-binary-tree/104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(!root)
            return 0;
        int a = maxDepth(root->left);
        int b = maxDepth(root->right);
        if(a!=0&&b!=0)
        return max(a,b)+1;
        if(a==0)
            return b+1;
         return a+1; 
    }
};

File: 104-maximum-depth-of-binary-tree/104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = 0;
    
    public int maxDepth(TreeNode root) {
        solve(root,0);
        return ans;
    }
    public void solve(TreeNode root , int d)
    {
        if(root == null)
        {
            ans = Math.max(ans,d);
            return ;
        }
        solve(root.left,d+1);
        solve(root.right,d+1);
    }
}